lib/pf/registration.pm

```diff
@@ -0,0 +1,154 @@
+package pf::registration;
+
+=head1 NAME
+
+pf::registration - node registration logic
+
+=cut
+
+=head1 DESCRIPTION
+
+pf::registration
+
+The module to place all node registration logic
+
+=cut
+
+use strict;
+use warnings;
+use pf::error;
+use pf::StatsD;
+use pf::StatsD::Timer;
+use pf::log;
+use pf::person;
+use pf::lookup::person;
+use pf::violation;
+use pf::util;
+use pf::dal::person;
+use pf::Portal::ProfileFactory; 
+use pf::constants::scan qw($SCAN_VID $POST_SCAN_VID);
+use pf::constants::parking qw($PARKING_VID);
+
+=head2 register_node
+
+register_node
+
+=cut
+
+sub register_node {
+    my $timer = pf::StatsD::Timer->new();
+    my ($node, $info) = @_;
+    my $logger = get_logger();
+    my $mac = $node->mac;
+    my $auto_registered = 0;
+
+    my $status_msg = "";
+    my $pid = $node->pid;
+
+    my ($status, $person) = pf::dal::person->find_or_create({"pid" => $pid});
+
+    if ($status == $STATUS::CREATED ) {
+        pf::lookup::person::async_lookup_person($pid, $info->{'source'});
+    }
+    if ($person) {
+        $person->source($info->{source});
+        $person->portal($info->{portal});
+        $person->save;
+    }
+
+    # if it's for auto-registration and mac is already registered, we are done
+    if ($info->{'autoreg'}) {
+        $node->autoreg('yes');
+        if ($node->status eq 'reg' ) {
+            $status = $node->save;
+            $status_msg = '';
+            if (is_error($status)) {
+                $status_msg = "modify of node $mac failed";
+                $logger->error($status_msg);
+            }
+            return ($status, $status_msg);
+        }
+    }
+    else {
+    # do not check for max_node if it's for auto-register
+        if ( is_max_reg_nodes_reached($mac, $pid, $node->category, $node->category_id) ) {
+            $status_msg = "max nodes per pid met or exceeded";
+            $logger->error( "$status_msg - registration of $mac to $pid failed" );
+            return ($STATUS::PRECONDITION_FAILED, $status_msg);
+        }
+    }
+
+    $node->status('reg');
+    $node->regdate(mysql_date());
+    $status = $node->save;
+
+    if (is_error($status)) {
+        $status_msg = "modify of node $mac failed";
+        $logger->error($status_msg);
+        return ($status, $status_msg);
+    }
+    $pf::StatsD::statsd->increment( called() . ".called" );
```

Comment:
Would be cleaner if you increment it at the top of the sub instead of the middle